### PR TITLE
Activate dual funding by default

### PR DESCRIPTION
**Pull request by t-bast**
And update the release notes.

### CODE REVIEW

**Review by remyers — CHANGES_REQUESTED** [review]
Looks good; just a few nits/typos in the release notes.

I couldn't find the explanation of the switch in `MinimalNodeFixture` from `DummyOnChainWallet` to `SingleKeyOnChainWallet`.

I can see that tests fail without that change, but some more information about the switch would make that change easier to review.

File: docs/release-notes/eclair-vnext.md [hunk 1]
```diff
@@ -4,6 +4,21 @@
...
 
+### Dual funding
+
+After many years of work and refining the protocol, [dual funding](https://github.com/lightning/bolts/pull/851) has been added to the BOLTs.
+
+This release of eclair activates this new feature, that will be used with [cln](https://github.com/ElementsProject/lightning/) nodes automatically. When opening channels to nodes that don't support dual funding, the older funding protocol will be used automatically.
+
+One of the immediate benefits of dual funding is that the funding transaction can now be RBF-ed, using the `rbfopen` RPC.
+
+There is currently no way to automatically add funds to channels that are being opened to your node, as deciding whether to do so or not really depends on each node operator's peering strategy. We are however working on plugin examples that node operators can fork to implement their own strategy for contributing to inbound channels.
```
[the remark this answers is not in the document]
Comment (t-bast) [reply docs/release-notes/eclair-vnext.md]: I linked to the plugin sample instead in https://github.com/ACINQ/eclair/pull/2825/commits/e551ecf7846470a040c1011f8b9b4762591c7c30

File: docs/release-notes/eclair-vnext.md [hunk 2]
```diff
@@ -4,6 +4,21 @@
 
 ## Major changes
 
+### Dual funding
+
+After many years of work and refining the protocol, [dual funding](https://github.com/lightning/bolts/pull/851) has been added to the BOLTs.
+
+This release of eclair activates this new feature, that will be used with [cln](https://github.com/ElementsProject/lightning/) nodes automatically. When opening channels to nodes that don't support dual funding, the older funding protocol will be used automatically.
```
[the remark this answers is not in the document]
Comment (t-bast) [reply docs/release-notes/eclair-vnext.md]: Thanks, updated in https://github.com/ACINQ/eclair/pull/2825/commits/e551ecf7846470a040c1011f8b9b4762591c7c30

**Comment by t-bast** [comment]
> I couldn't find the explanation of the switch in MinimalNodeFixture from DummyOnChainWallet to SingleKeyOnChainWallet.

Right, that's because the `DummyOnChainWallet` is overriding the `makeFundingTx` function which is used for single-funded channels, but its override of the `fundTransaction` function that is used for dual-funding is basically a no-op that returns the transaction as-is. This breaks dual funding entirely because it doesn't really return a funded transaction.

The `SingleKeyOnChainWallet` is a better mock for an on-chain wallet that supports funding transactions, which makes it work for dual funding tests.

**Pull request merged**